staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1258,7 +1258,7 @@ func (c *cacheWatcher) add(event *watchCacheEvent, timer *time.Timer) bool {
 		// This means that we couldn't send event to that watcher.
 		// Since we don't want to block on it infinitely,
 		// we simply terminate it.
-		klog.V(1).Infof("Forcing %v watcher close due to unresponsiveness: %v", c.objectType.String(), c.identifier)
+		klog.V(1).Infof("[%p] Forcing %v watcher close due to unresponsiveness: %v. len(c.input) = %v, len(c.result) = %v", c, c.objectType.String(), c.identifier, len(c.input), len(c.result))
```

Comment:
I think that this is universally good - so let's extract this one to a separate PR and have that merge.

[The other changes may potentially slightly affect performance, so I would like to first see their impact]

Reply:
Please see https://github.com/kubernetes/kubernetes/pull/98208